### PR TITLE
checkra1n: init at 0.12.4

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change
Add [checkra1n](https://checkra.in/), Jailbreak tool for iPhone and other Apple devices, at version 0.12.4.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/checkra1n/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "Jailbreak for iPhone 5s though iPhone X, iOS 12.0 and up";
+    homepage = "https://checkra.in/";
+    license = licenses.unfreeRedistributable;
+    maintainers = with maintainers; [ onny ];
+    platforms = platforms.linux;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/mobile/checkra1n/default.nix:27]: One final new line to much.

File: pkgs/development/mobile/checkra1n/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "checkra1n";
+  version = "0.12.4";
+
+  src = fetchurl {
+    url = "https://assets.checkra.in/downloads/linux/cli/x86_64/dac9968939ea6e6bfbdedeb41d7e2579c4711dc2c5083f91dced66ca397dc51d/checkra1n";
```
Comment (SuperSandro2000) [inline pkgs/development/mobile/checkra1n/default.nix:8]: I don't like that there is no versioned download or update script.

Comment (onny) [reply pkgs/development/mobile/checkra1n/default.nix:8]: I'll see if they provide versioned downloads, but I'm not sure

Comment (onny) [reply pkgs/development/mobile/checkra1n/default.nix:8]: Looks like the checksum represents the specific version / release. They don't have any other binary distribution :(

**Review by splawless — APPROVED** [review]
[**_[@[@](https://github.com/NixOS/nixpkgs/pull/128798/files)](https://github.com/NixOS/nixpkgs/pull/128798/files)_**](https://github.com/NixOS/nixpkgs/pull/128798/files)

**Pull request merged**